### ISSUE TITLE
[win][fs] `path.toNamespacedPath` does nothing

### ISSUE THREAD

**Issue by AlttiRi** [oven-sh/bun#8247]
### What version of Bun is running?

1.0.23

### What platform is your computer?

Window 10

### What steps can reproduce the bug?

```js
import path from "node:path";

const fp1 = "C:\\Downloads\\bun-windows-x64\\bun.exe";
const fp2 = "bun.exe";
const fp3 = "\\\\KEENETIC\\USB";
const fp4 = "\\\\KEENETIC\\USB\\A";
console.log(path.toNamespacedPath(fp1));
console.log(path.toNamespacedPath(fp2));
console.log(path.toNamespacedPath(fp2) === "\\\\?\\" + path.resolve(fp2));
console.log(path.toNamespacedPath(fp3));
console.log(path.toNamespacedPath(fp4));
```

### What is the expected behavior?

```
\\?\C:\Downloads\bun-windows-x64\bun.exe
\\?\C:\Downloads\bun-windows-x64\bun.exe
true                               
\\?\UNC\KEENETIC\USB\
\\?\UNC\KEENETIC\USB\A
```

### What do you see instead?

```
C:\Downloads\bun-windows-x64\bun.exe
bun.exe
false
\\KEENETIC\USB
\\KEENETIC\USB\A
```

### Additional information

Rel: 
- https://github.com/oven-sh/bun/issues/8246
- https://github.com/oven-sh/bun/issues/8836
- https://github.com/oven-sh/bun/issues/8248

**Comment by jdalton** [comment]
Context:
https://nodejs.org/api/path.html#pathtonamespacedpathpath
https://github.com/oven-sh/bun/blob/c21dc25a6546f0b74af62f4826fcf2a494805882/src/bun.js/bindings/Path.cpp#L125-L127

See zig's [wToPrefixedFileW](https://github.com/ziglang/zig/blob/9e684e8d1af39904055abe64a9afda69a3d44a59/lib/std/os/windows.zig#L2223C8-L2223C24)

**Comment by paperclover** [comment]
Fixed in #8469
